includes/REST_API/Stories_Taxonomies_Controller.php

```diff
@@ -83,6 +84,73 @@ public function prepare_item_for_response( $taxonomy, $request ): WP_REST_Respon
 		return apply_filters( 'rest_prepare_taxonomy', $response, $taxonomy, $request );
 	}
 
+	/**
+	 * Retrieves all public taxonomies.
+	 *
+	 * Adds support for filtering by the hierarchical attribute.
+	 *
+	 * @since 1.22.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_items( $request ) {
+		// Retrieve the list of registered collection query parameters.
+		$registered = $this->get_collection_params();
+
+		if ( isset( $registered['type'] ) && ! empty( $request['type'] ) ) {
+			$taxonomies = get_object_taxonomies( $request['type'], 'objects' );
+		} else {
+			$taxonomies = get_taxonomies( '', 'objects' );
+		}
+
+		if ( isset( $registered['hierarchical'], $request['hierarchical'] ) ) {
+			$taxonomies = wp_filter_object_list( $taxonomies, [ 'hierarchical' => (bool) $request['hierarchical'] ] );
+		}
+
+		$data = [];
+
+		foreach ( $taxonomies as $tax_type => $value ) {
+			if ( empty( $value->show_in_rest ) || ( 'edit' === $request['context'] && ! current_user_can( $value->cap->assign_terms ) ) ) {
+				continue;
+			}
+
+			$tax               = $this->prepare_item_for_response( $value, $request );
+			$tax               = $this->prepare_response_for_collection( $tax );
+			$data[ $tax_type ] = $tax;
```

Comment:
I hate the shape of this response. If we are doing this, can we change the shape of this.

Reply:
```suggestion
			$data[] = $tax;
```

Reply:
This is just copied from core and the expected response format

AFAIK we also rely on this in the editor, so I would not wanna change this.

![Screenshot 2022-06-08 at 18 05 05](https://user-images.githubusercontent.com/841956/172664437-0348a962-fad2-40f7-b8cc-ee58214eb867.png)